### PR TITLE
REGR: MultiIndex.join does not work for ea dtypes

### LINKED ISSUE #49277: BUG: inner join with StringDtype multi-indexes gives unexpected result in 1.5.x
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

x = pd.DataFrame({"a": ["a1", "a1", "a2"], "b": ["b1", "b2", "b3"]}, dtype=pd.StringDtype())
x = x.set_index(["a", "b"])
y = pd.DataFrame({"a": ["a1"], "c": ["c1"]}, dtype=pd.StringDtype())
y = y.set_index(["a", "c"])
z = x.join(y, how="inner")
```

### Issue Description

When joining 2 data frames with the StringDtype dtype index, the inner join gives incorrect results. The outcome of z returns a multi-index with only 1 row:
` "a1", "b1", "c1"`
It does work when I don't explicitly set the dtype. In that case the index dtypes are object.

### Expected Behavior

The expected result for data frame z is 2 rows (just like in Pandas 1.4.x), namely:
```
"a1", "b1", "c1"
"a1", "b2", "c1"
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 91111fd99898d9dcaa6bf6bedb662db4108da6e6
python           : 3.10.8.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19044
machine          : AMD64
processor        : Intel64 Family 6 Model 85 Stepping 7, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United States.1252
pandas           : 1.5.1
numpy            : 1.23.4
pytz             : 2022.5
dateutil         : 2.8.2
setuptools       : 65.3.0
pip              : 22.3
Cython           : None
pytest           : 7.1.3
hypothesis       : None
sphinx           : 5.3.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : 1.0.9
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : 1.4.42
tables           : None
tabulate         : 0.8.10
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #49277 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

This can not fall through for regular ea dtypes, rather go through object path

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -4636,8 +4636,9 @@ def join(
             return self._join_non_unique(other, how=how)
         elif not self.is_unique or not other.is_unique:
             if self.is_monotonic_increasing and other.is_monotonic_increasing:
-                if self._can_use_libjoin:
+                if not is_interval_dtype(self.dtype):
```
Comment (mroeschke) [inline pandas/core/indexes/base.py:4639]: So `_can_use_libjoin` shouldn't be updated instead?

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -4636,8 +4636,9 @@ def join(
...
                     # otherwise we will fall through to _join_via_get_indexer
+                    # go through object dtype for ea till engine is supported properly
```
Comment (mroeschke) [inline pandas/core/indexes/base.py:4641]: Do we have a GH issue related to an engine for EAs?

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (phofl) [reply pandas/core/indexes/base.py:4639]: No, in most cases going through object dtype is slower than the alternative. In our case here the alternative can not handle non interval dtype

File: pandas/core/indexes/base.py (hunk 2 above)
Comment (phofl) [reply pandas/core/indexes/base.py:4641]: #39133

**Review by mroeschke — APPROVED** [review]

**Comment by phofl** [comment]
good to merge @mroeschke ?

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**